### PR TITLE
[Windows] Adding property in installer to support driver rollback

### PR DESCRIPTION
**Pull request by mrafi97**
### What does this PR do?
Adds the properties `DDDRIVERROLLBACK_NPM` and `DDDRIVERROLLBACK_PROCMON` to allow for rollback of `ddnpm` and `ddprocmon` drivers after a failed install/upgrade. 

To account for prior agent versions not having the uninstall rollback action, the installer will view previous and current versions and only perform rollbacks for newer agent versions.

### Motivation
https://datadoghq.atlassian.net/browse/WINA-726

### Describe how to test/QA your changes
1. Run the Datadog Agent MSI with the property `WIXFAILWHENDEFERRED=1` to induce an failed install/upgrade:
```
Start-Process -Wait msiexec -ArgumentList '/qn /i datadog-agent*.msi APIKEY="APIKEY" WIXFAILWHENDEFERRED=1'
```
2. Check running services:
```
Get-Service "datadog*" 
Get-Service "ddnpm" 
Get-Service "ddprocmon"
```
Expected results for failed install/upgrade:
- Fresh install: All services are removed
- Upgrade from 7.52, 7.54, 7.55: All services are still installed
- Upgrade from < 7.52: Only `ddprocmon` is removed

### CODE REVIEW

File: releasenotes/notes/windows-driver-rollback-36c4de3614d988b9.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Windows: Added driver rollback properties to ensure that all services/drivers are uninstalled/rolled back after a failed installation/upgrade.
```
Comment (alai97) [inline releasenotes/notes/windows-driver-rollback-36c4de3614d988b9.yaml:11]:
```suggestion
    Windows: Added driver rollback properties to ensure that all services and drivers are uninstalled or rolled back after an installation or upgrade failure.
```

**Review by alai97 — APPROVED** [review]
Looks good, one suggestion.

**Review by vickenty — APPROVED** [review]

File: tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs [hunk 2]
```diff
@@ -154,6 +160,61 @@ public void GetWindowsBuildVersion()
...
+                }
+                else
+                {
+                    version_53 = new Version("6.53");
+                    minimumVersion = new Version("6.56");
+                }
+
+                var compareResult = currentVersion.CompareTo(minimumVersion);
+                if (compareResult < 0) // currentVersion is less than minimumVersion
+                {
```
Comment (clarkb7) [inline tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs:190]: Use a more descriptive comment
```suggestion
                if (compareResult < 0)
                {
                    // upgrading from a version that didn't implement driver rollback
                    // Clear NPM flag to ensure NPM service is not deleted on rollback
```

File: tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs [hunk 3]
```diff
@@ -154,6 +160,61 @@ public void GetWindowsBuildVersion()
...
+                    _session["DDDRIVERROLLBACK_NPM"] = "";
+
+                    var compare_53 = currentVersion.CompareTo(version_53);
+                    if (compare_53 < 0) //currentVersion is less than 6.53/7.53
+                    {
```
Comment (clarkb7) [inline tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs:195]:
```suggestion
                    if (compare_53 < 0) //currentVersion is less than 6.53/7.53
                    {
                        // upgrading from a version that did not include procmon
                        // Set PROCMON flag to ensure PROCMON service is deleted on rollback
```

File: tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs [hunk 4]
```diff
@@ -154,6 +160,61 @@ public void GetWindowsBuildVersion()
...
+                        _session["DDDRIVERROLLBACK_PROCMON"] = "";
+                    }
+                }
+                else // currentVersion is not less than minimumVersion
+                {
+                    _session["DDDRIVERROLLBACK_NPM"] = "1";
+                    _session["DDDRIVERROLLBACK_PROCMON"] = "1";
+                }
+            }
+            else  // This is a fresh install
```
Comment (clarkb7) [inline tools/windows/DatadogAgentInstaller/CustomActions/InstallStateCustomActions.cs:209]:
```suggestion
            else  // This is a fresh install, set rollback flags to ensure drivers are removed on rollback
```

**Review by clarkb7 — APPROVED** [review]

**Pull request merged**